### PR TITLE
md4c: conan v2 support

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **md4c/0.4.8**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
the test_package.cpp or test_package.c is missing

**Comment by toge** [comment]
@uilianries 
Exactly! 
It's my mistake.
Thank you for notice!

**Review by jgsogo — APPROVED** [review]

File: recipes/md4c/all/conanfile.py [hunk 1]
```diff
@@ -22,62 +25,76 @@ class Md4cConan(ConanFile):
...
+        tc = CMakeToolchain(self)
         if self.options.encoding == "utf-16":
-            cmake.definitions["CONAN_C_FLAGS"] = "-DMD4C_USE_UTF16"
+            tc.cache_variables["CONAN_C_FLAGS"] = "-DMD4C_USE_UTF16"
         elif self.options.encoding == "ascii":
-            cmake.definitions["CONAN_C_FLAGS"] = "-DMD4C_USE_ASCII"
-
-        cmake.configure()
-        return cmake
+            tc.cache_variables["CONAN_C_FLAGS"] = "-DMD4C_USE_ASCII"
```
Comment (SpaceIm) [inline recipes/md4c/all/conanfile.py:68]: What does it mean for CMakeToolchain? Nothing I guess.

File: recipes/md4c/all/conandata.yml [hunk 2]
```diff
@@ -5,5 +5,5 @@ sources:
 
 patches:
   "0.4.8":
-    - base_path: "source_subfolder"
-      patch_file: "patches/0.4.8-0001-remove-unused-files.patch"
+    - patch_file: "patches/0.4.8-0001-disable-build-shared-flag.patch"
+      patch_description: "disable setting for BUILD_SHARED_LIBS"
```
Comment (SpaceIm) [inline recipes/md4c/all/conandata.yml:9]: this patch is not needed, inject CMP0077 externally instead.

**Comment by SpaceIm** [comment]
These lines in upstream CMakeLists are suspicious and should be removed: https://github.com/mity/md4c/blob/release-0.4.8/CMakeLists.txt#L45-L53

File: recipes/md4c/all/conanfile.py [hunk 3]
```diff
@@ -22,62 +25,76 @@ class Md4cConan(ConanFile):
...
         self.cpp_info.libs = ["md4c", "md4c-html",]
 
         if self.options.encoding == "utf-16":
             self.cpp_info.defines.append("MD4C_USE_UTF16")
+        elif self.options.encoding == "ascii":
+            self.cpp_info.defines.append("MD4C_USE_ASCII")
+
+        self.cpp_info.set_property("cmake_file_name", "md4c")
+        self.cpp_info.set_property("cmake_target_name", "md4c::md4c")
+        self.cpp_info.set_property("pkg_config_name", "md4c")
```
Comment (SpaceIm) [inline recipes/md4c/all/conanfile.py:95]: There are 2 different targets and pc files, one for each lib: https://github.com/mity/md4c/blob/release-0.4.8/src/CMakeLists.txt#L34-L55

Moreover md4c-html depends on md4c, not the opposite.

**Pull request merged**